Orange/widgets/io.py

```diff
@@ -27,27 +27,32 @@ def _get_exporter():
         pass
 
     @staticmethod
+    @abstract
     def _export(self, exporter, filename):
-        raise NotImplementedError(
-            "Descendants of ImgFormat must override method _export")
+        pass
 
     @classmethod
     def write_image(cls, filename, scene):
-        from pyqtgraph.graphicsItems.GraphicsWidget import GraphicsWidget
-
         try:
             exporter = cls._get_exporter()
             cls._export(exporter(scene), filename)
-        except:
-            source = scene.sceneRect().adjusted(-15, -15, 15, 15)
-            buffer = cls._get_buffer(source.size(), filename)
+        except Exception:
+            try:
+                rect = scene.sceneRect()
+            except AttributeError:  # not a QGraphicsView/Scene; just a QWidget
+                rect = scene.rect()
+            rect = rect.adjusted(-15, -15, 15, 15)
+            buffer = cls._get_buffer(rect.size(), filename)
 
             painter = QtGui.QPainter()
             painter.begin(buffer)
             painter.setRenderHint(QtGui.QPainter.Antialiasing)
 
-            target = cls._get_target(scene, painter, buffer, source)
-            scene.render(painter, target, source)
+            target = cls._get_target(scene, painter, buffer, rect)
+            try:
+                scene.render(painter, target, rect)
+            except TypeError:
+                scene.render(painter)  # PyQt4 QWidget.render() takes different params
```

Comment:
I don't find [it](https://github.com/biolab/orange3/pull/1034/files) much better now. Other blocks were left for generality.